osidb/models.py

```diff
@@ -1894,6 +1894,36 @@ class Meta:
     def __str__(self):
         return str(self.uuid)
 
+    def _validate_tracker_affect(self):
+        """
+        check that the tracker is associated with an affect
+        """
+        # there are no references before the first save to DB
+        if self._state.adding:
+            return
+
+        if not self.affects.first():
+            raise ValidationError("Tracker must be associated with an affect")
+
+    def _validate_tracker_ps_module(self):
+        """
+        check that the tracker is associated with a valid PS module
+        """
+        if not self.affects.first():
+            return
+
+        if not PsModule.objects.filter(name=self.affects.first().ps_module):
+            raise ValidationError("Tracker must be associated with a valid PS module")
```

Comment:
This check is weird, `affects` is a many relationship, so how do we actually know whether a `Tracker` is associated with a valid PS module?

In most cases tracker<->affect is a 1:1 relationship but what happens if there's more than one affect per tracker? Should we only consider the first affect? What if the other affects DO have a valid `PsModule`?

Plus IMO this validation should be done at the affect level no? It would simplify the check.

Reply:
@Elkasitu having multiple associated affects is a valid situation but then they all must have the same PS module and component which is being checked by `_validate_multi_flaw_tracker`. Therefore we can here simply take the first. 

There are validations on the `Affect` model checking whether it has a valid PS module. However, here it is necessary to be checked too as it is a requirement to be able to create/update a tracker and when doing it the affect validations are not run as the affect is not being changed. So here it is clearly a safeguard before the work is given to the `tracker` app.